core/field_label.js

```diff
@@ -38,18 +38,30 @@ goog.require('Blockly.utils.Size');
  * Class for a non-editable, non-serializable text field.
  * @param {string=} opt_value The initial value of the field. Should cast to a
  *    string. Defaults to an empty string if null or undefined.
- * @param {string=} opt_class Optional CSS class for the field's text.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/label#creation}
+ *    for a list of properties this parameter supports.
  * @extends {Blockly.Field}
  * @constructor
  */
-Blockly.FieldLabel = function(opt_value, opt_class) {
+Blockly.FieldLabel = function(opt_value, opt_config) {
   this.size_ = new Blockly.utils.Size(0, Blockly.Field.TEXT_DEFAULT_HEIGHT);
-  this.class_ = opt_class;
+
+  /**
+   * The css class name to apply to the field's textElement_.
+   * @type {?string}
+   * @private
+   */
+  this.class_ = null;
+
   opt_value = this.doClassValidation_(opt_value);
   if (opt_value === null) {
     opt_value = '';
   }
-  this.setValue(opt_value);
+  Blockly.FieldLabel.superClass_.constructor.call(
+      this, opt_value, null, opt_config);
```

Comment:
this.size_ is going to be overridden by the parent constructor with this call.
What's the reason for ``doClassValidation_`` going before the parent constructor? If none, then we can easily move this to the top, so that configuration inheritance is achieved.

Reply:
Class validation must be handled before passing the value to the constructor. Otherwise if the developer passes an invalid value to the constructor or JSON the `value_` ends up being null.

This is the course of action recommended in the [docs](https://developers.google.com/blockly/guides/create-custom-blocks/fields/customizing-fields/creating).

Moved this.size_ set to after the super-constructor call.

Reply:
Sounds good. Thanks for moving that. Let's try: 
- super class (as close as possible to the top)
- initializations (this.class, this.size)
- finally the configure call.

Reply:
Fixed!